sql/expression/function/json_array.go

```diff
@@ -0,0 +1,143 @@
+// Copyright 2022 Dolthub, Inc.
+//
+// Licensed under the Apache License, Version 2.0 (the "License");
+// you may not use this file except in compliance with the License.
+// You may obtain a copy of the License at
+//
+//     http://www.apache.org/licenses/LICENSE-2.0
+//
+// Unless required by applicable law or agreed to in writing, software
+// distributed under the License is distributed on an "AS IS" BASIS,
+// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+// See the License for the specific language governing permissions and
+// limitations under the License.
+
+package function
+
+import (
+	"fmt"
+	"strings"
+	"time"
+
+	"github.com/dolthub/go-mysql-server/sql"
+)
+
+// JSON_ARRAY([val[, val] ...])
+//
+// JSONArray Evaluates a (possibly empty) list of values and returns a JSON array containing those values.
+//
+// https://dev.mysql.com/doc/refman/8.0/en/json-creation-functions.html#function_json-array
+
+type JSONArray struct {
+	Docs []sql.Expression
```

Comment:
1. No reason to export this
2. `Docs` is not the right name. This function does not necessarily have to take in a json doc. Use `vals` or `args` or something more expressive.